gcloud/bigtable/column_family.py

```diff
@@ -15,6 +15,59 @@
 """User friendly container for Google Cloud Bigtable Column Family."""
 
 
+class GarbageCollectionRule(object):
+    """Table garbage collection rule.
+
+    Cells in the table fitting the rule will be deleted during
+    garbage collection.
+
+    .. note::
+
+        At most one of ``max_num_versions`` and ``max_age`` can be specified
+        at once.
```

Comment:
I never thought about it, it may be a good idea! This follows a pattern also present in the `RowFilter` class, but that one has 15 or so properties so breaking it into 15 subclasses may be a bit over the top. Let me grab you links (on my phone so internet is difficult to use) so you can see what this is building towards.